packages/trpc/server/routers/viewer/apiKeys/create.handler.ts

```diff
@@ -17,12 +20,38 @@ export const createHandler = async ({ ctx, input }: CreateHandlerOptions) => {
   const [hashedApiKey, apiKey] = generateUniqueAPIKey();
 
   // Here we snap never expires before deleting it so it's not passed to prisma create call.
-  const { neverExpires, ...rest } = input;
+  const { neverExpires, teamId, ...rest } = input;
+
+  if (teamId) {
+    const user = await prisma.user.findFirst({
+      where: {
+        id: ctx.user.id,
+      },
+      include: {
+        teams: true,
+      },
+    });
+
+    const userHasAdminOwnerPermissionInTeam =
+      user &&
+      user.teams.some(
+        (membership) =>
+          membership.teamId === teamId &&
+          (membership.role === MembershipRole.ADMIN || membership.role === MembershipRole.OWNER)
+      );
+
+    if (!userHasAdminOwnerPermissionInTeam) {
```

Comment:
only admin or owner can create apiKeys of team ( if teamId is passed)

Reply:
Refactoring this to get this from the query